can/interfaces/usb2can/usb2canInterface.py

```diff
@@ -1,132 +1,132 @@
-        will be used to identify the first such device. The *kwarg* `serial` may also be
-        used.
+        will be used to identify the first such device.
 
-    :param int bitrate:
+    :param int bitrate (optional):
         Bitrate of channel in bit/s. Values will be limited to a maximum of 1000 Kb/s.
         Default is 500 Kbs
 
-    :param int flags:
+    :param int flags (optional):
         Flags to directly pass to open function of the usb2can abstraction layer.
+
+    :param str dll (optional):
+        Path to the DLL with the CANAL API to load
+        Defaults to 'usb2can.dll'
+
+    :param str serial (optional):
+        Alias for `channel` that is provided for legacy reasons.
+        If both `serial` and `channel` are set, `serial` will be used and
+        channel will be ignored.
+
     """
 
-    def __init__(self, channel, *args, **kwargs):
+    def __init__(self, channel=None, dll="usb2can.dll", flags=0x00000008,
+                 bitrate=500000, *args, **kwargs):
 
-        self.can = Usb2CanAbstractionLayer()
+        self.can = Usb2CanAbstractionLayer(dll)
 
-        # set flags on the connection
-        if 'flags' in kwargs:
-            enable_flags = kwargs["flags"]
+        # get the serial number of the device
+        if "serial" in kwargs:
+            device_id = kwargs["serial"]
         else:
-            enable_flags = 0x00000008
+            device_id = channel
 
-        # code to get the serial number of the device
-        if 'serial' in kwargs:
-            deviceID = kwargs["serial"]
-        elif channel is not None:
-            deviceID = channel
-        else:
-            from can.interfaces.usb2can.serial_selector import serial
-            deviceID = serial()
+        # search for a serial number if the device_id is None or empty
+        if not device_id:
+            devices = find_serial_devices()
+            if not devices:
+                raise CanError("could not automatically find any device")
+            device_id = devices[0]
 
-        # get baudrate in b/s from bitrate or use default
-        bitrate = kwargs.get("bitrate", 500000)
-        # convert to kb/s (eg:500000 bitrate must be 500), max rate is 1000 kb/s
-        baudrate = min(1000, int(bitrate/1000))
+        # convert to kb/s and cap: max rate is 1000 kb/s
+        baudrate = min(int(bitrate // 1000), 1000)
 
-        connector = format_connection_string(deviceID, baudrate)
+        self.channel_info = "USB2CAN device {}".format(device_id)
 
-        self.handle = self.can.open(connector.encode('utf-8'), enable_flags)
+        connector = "{}; {}".format(device_id, baudrate)
+        # we need to convert this into bytes, since the underlying DLL cannot
+        # handle non-ASCII configuration strings
+        self.handle = self.can.open(connector, flags)
 
-        super(Usb2canBus, self).__init__(channel=channel, *args, **kwargs)
+        super(Usb2canBus, self).__init__(channel=channel, dll=dll, flags=flags,
```

Comment:
Why pass the `dll` and `flags` parameters to the parent class constructor? They are truly specific to this backend.

I could understand if the `BusABC` class somehow saved the config for later reference, but it doesn't.

Reply:
It doesn't hurt though, so I'd rather leave it there.